src/RegistrationMsg.chpl

```diff
@@ -496,7 +508,10 @@ module RegistrationMsg
                 // Convert the string back into an array for looping
                 var nameList = nameStr.split("+");
                 forall n in nameList with (+ reduce status) {
-                    var resp = unregisterMsg(cmd, n, st);
+                    var base_json = msgArgs.get("name").asMap();
+                    base_json.set("val", n);
```

Comment:
Shadow variables are talked about a bit more [here](https://chapel-lang.org/docs/primers/forallLoops.html?highlight=intents#task-intents-and-shadow-variables)

Reply:
I didn't even think to try `with`, I will try that now

Reply:
The format above should work based on Chapel's documentation. However, this results in the infamous Chpale compiler error `It's us, not you`. @pierce314159 believes this is due to `asMap` being a `throwing` function. However, defining the `base_json` variable outside of the loop and then using 
```chapel
with (in base_json, + reduce status)
```
should be more efficient than what is there and avoids the `throwing` function call from within the `with`.

Reply:
Just a quick update, I pulled down the PR and made `asMap` non-throwing, but got the same `It's us, not you` error. So that was not the cause of the issue. The `in` version seems to work tho, nice job!